### PR TITLE
Updated the "Generated an index" example to include more platforms.

### PR DESCRIPTION
**Pull request by tnabtaf**
Addresses issue #4601.

### Description

As documented in issue #4601 I had issues following the 'build a channel index' example in the docs.  This PR adds support for Windows and the new Apple Silicon to the list.  Also adds a section about how to get `wget` if you don't already have it.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Review by travishathaway — APPROVED** [review]

File: docs/source/concepts/generating-index.rst [hunk 1]
```diff
@@ -126,20 +126,62 @@ Example: Building a channel
...
-      $ mkdir linux-64 osx-64
 
-#. Put your favorite package in the channel.
+#. Install ``wget`` if you don't already have it.
+
+    Check if you have it.
+
+    .. code-block:: bash
+
+      $ which wget
```
Comment (kathatherine) [inline docs/source/concepts/generating-index.rst:142]: "which" is only a unix command, so this method is only available to mac and Linux users. 

I believe `where.exe wget` would be the Windows equivalent.

File: docs/source/concepts/generating-index.rst [hunk 2]
```diff
@@ -110,7 +110,7 @@ For each subdir:
 
 * Remove all packages that need to be removed.
 
-For all packages that need to be added/updated:
+* For all packages that need to be added/updated:
 
   * Extract the package to access metadata including full package name,
```
Comment (kathatherine) [inline docs/source/concepts/generating-index.rst:115]: It'd be great to add a comma here after "metadata", too.

**Review by kathatherine — CHANGES_REQUESTED** [review]
Couple things. Thanks for looking into this, Dave!

File: docs/source/concepts/generating-index.rst (hunk 1 above)
Comment (tnabtaf) [reply docs/source/concepts/generating-index.rst:142]: Thanks Katherine.  I had assumed that any Windows users will be using Windows Subsystem for Linux. I'll add this.

Comment (tnabtaf) [reply docs/source/concepts/generating-index.rst]: Updated. But this is dangerous as I don't understand windows, and don't have access to a windows machine.

Comment (kathatherine) [reply docs/source/concepts/generating-index.rst]: I have a vanilla Windows VM, so I attempted some testing. A few things:

- I can't get `conda install wget` to work. I had to pip install it.
- Once I did get it installed in an environment, `where wget`, `where.exe wget`, and `where wget.*` all give me the same thing: `INFO: Could not find files for the given pattern(s).`

So, seems like vanilla Windows is going to be a bit more complicated than everything Unix based.

**Review by kathatherine — APPROVED** [review]
Thanks so much for this, Dave! I trust you tested it as much as you were able, so approved!

**Comment by tnabtaf** [comment]
@kathatherine I tested this on a clean Windows 10 install.

**Review by jezdez — APPROVED** [review]

**Pull request merged**